### DISCUSSION TITLE
Cannot run FTP and SFTP

### DISCUSSION THREAD

**Discussion by fmomteiro** [drakkan/sftpgo#1109] [General]
Hello,

I'm having some trouble trying to run FTP and SFTP on Windows. 

If I set FTP to port 21 and SFTP to port 22, FTP does not start and SFTP starts on port 2022. 

If I set FTP port parameter to zero (disabling FTP), SFTP starts on port 22 as expected. 

My conf that causes this problem: 

`
  "sftpd": {
    "bindings": [
      {
        "port": 22,
        "address": "",
        "apply_proxy_config": true
      }
    ],
(...)
"ftpd": {
    "bindings": [
      {
        "port": 21,
        "address": "",
        "apply_proxy_config": true,
        "tls_mode": 0,
        "certificate_file": "",
        "certificate_key_file": "",
        "min_tls_version": 12,
        "force_passive_ip": "",
        "passive_ip_overrides": [],
        "client_auth_type": 0,
        "tls_cipher_suites": [],
        "passive_connections_security": 0,
        "active_connections_security": 0,
        "debug": false
      }
    ],
`

Log:

`
{"level":"info","time":"2022-12-18T23:24:37.618","sender":"service","message":"FTP server not started, disabled in config file"}
(...)
{"level":"info","time":"2022-12-18T23:24:37.618","sender":"service","message":"initializing SFTP server with config {Banner:SFTPGo_2.4.2 Bindings:[{Address: Port:2022 ApplyProxyConfig:true}] MaxAuthTries:0 HostKeys:[] HostCertificates:[] HostKeyAlgorithms:[] Moduli:[] KexAlgorithms:[] Ciphers:[] MACs:[] TrustedUserCAKeys:[] RevokedUserCertsFile: LoginBannerFile: EnabledSSHCommands:[md5sum sha1sum sha256sum cd pwd scp] KeyboardInteractiveAuthentication:false KeyboardInteractiveHook: PasswordAuthentication:true FolderPrefix: certChecker:<nil> parsedUserCAKeys:[]}"}
`

Could it be a misconfiguration?

**Comment by drakkan** [comment]
Hi,

check your logs to understand if your configuration file is correctly parsed. I guess after your modification is not a valid JSON anymore and so the default configuration is used
